src/components/upload/video-uploader.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react'
+import ReactS3Uploader from 'react-s3-uploader'
+import {getAuthorizationHeader} from 'utils/auth'
+import uuid from 'shortid'
+import fileExtension from 'file-extension'
+
+const SIGNING_URL = `/api/aws/sign-s3`
+
+const VideoUploader = ({
```

Comment:
I moved the React S3 Uploader into its own component file. Otherwise it was buried in a sea of JSX and that made it hard to get to and reason about the important logic that it does.

Reply:
Great change. Looks a lot nicer